src/main.rs

```diff
@@ -93,38 +131,40 @@ fn main() {
         process_list.update().expect("Can't update process list");
         check_apps_and_kill(&apps, &process_list.processes);
 
-        // going for a short nap (500ms)
-        thread::sleep(Duration::from_millis(500));
+        thread::sleep(Duration::from_millis(args.delay));
     }
 }
 
 fn check_apps_and_kill(
     apps: &Vec<App>,
     process_list: &BTreeMap<psutil::Pid, psutil::process::Process>,
 ) {
-    let pname_pid: Vec<_> = process_list
-        .iter()
-        .map(|x| (x.0, x.1.name().unwrap_or("".into())))
-        .collect();
-
     let now = chrono::Local::now().time();
+    for process in process_list {
+        let process = process.1;
 
-    for app in apps {
-        if let Some(process) = pname_pid.iter().find(|&p| p.1 == app.command) {
-            if app.enabled && kill_or_not(&app, &now) {
-                println!("killing {}", app.name);
-                Command::new("kill")
-                    .args(&["-9", &process.0.to_string()])
-                    .output()
-                    .expect(&format!(
-                        "failed to kill process {}, with PID {}",
-                        &(app.command),
-                        process.0,
-                    ));
+        if let Ok(Some(cmd)) = process.cmdline() {
+            let cmd = cmd.split(" ").collect::<Vec<&str>>();
+            for app in apps {
+                if app.command == cmd[0] {
+                    if (app.args.is_empty() || check_args(&app.args, &cmd[1..].join(" ")))
+                        && (app.enabled && kill_or_not(&app, &now))
+                    {
+                        println!("killing {}", app.name);
+                        process.kill().expect("Failed to kill process");
+                    }
+                }
             }
         }
     }
 }
+fn check_args(user_args: &str, args: &str) -> bool {
+    if user_args.contains("*") {
+        lazy_regex!(&user_args.replace("*", ".*")).is_match(args)
```

Comment:
Why are you replacing `.*` which is a valid regex expression with `*` ? This way people may get confused.
```suggestion
        lazy_regex!(&user_args).is_match(args)
```

Reply:
because they can use an evil regex(without even knowing) and hang the program, so it won't be able to kill programs because it's checking for arguments and they think it is the program's fault.
and i think most of people won't make this mistake because it's like globing in bash

https://owasp.org/www-community/attacks/Regular_expression_Denial_of_Service_-_ReDoS